### PR TITLE
Including OCI server tests, no OCI mocks yet

### PR DESCRIPTION
**Pull request by chGoodchild**
This pull request includes the tests from `chGoodchild:use_zchunk` as well as OCI tests...

### CODE REVIEW

File: environment.yml [hunk 1]
```diff
@@ -20,3 +20,4 @@ dependencies:
   - pyyaml
   - pip:
     - pytest-xprocess
+    - pygit2
```
Comment (wolfv) [inline environment.yml:23]: do we still need this?

File: src/mirrors/oci.cpp [hunk 2]
```diff
@@ -188,7 +188,7 @@ namespace powerloader
             checksum = &data->sha256sum;
         }
         auto [split_path, split_tag] = split_path_tag(target->target->path);
-
+        std::cout << "format1" << std::endl;
```
Comment (wolfv) [inline src/mirrors/oci.cpp:191]: can be removed, right?

File: src/uploader/oci_upload.cpp [hunk 3]
```diff
@@ -54,6 +54,8 @@ namespace powerloader
         std::string temp_upload_location = response.header["location"];
 
         auto upload_url = format_upload_url(mirror.url, temp_upload_location, digest);
+        // std::string upload_url = fmt::format("{}{}?digest={}", mirror.url, temp_upload_location,
```
Comment (wolfv) [inline src/uploader/oci_upload.cpp:57]: ? maybe remove?

File: test/remote_mirrors.yml [hunk 4]
```diff
@@ -2,6 +2,7 @@ mirrors:
   conda-forge:
     - https://conda.anaconda.org/conda-forge
     - https://repo.mamba.pm/conda-forge
+    - https://beta.mamba.pm/get/conda-forge
```
Comment (wolfv) [inline test/remote_mirrors.yml:5]: repo.mamba.pm and beta.mamba.pm/get... point to the same thing. I am happy to keep this, just wondering what the motivation was :)

File: test/test_OCImain_branch.py [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+        # Delete the file locally
+        # Path(upload_path).unlink()
+
+        # Generate yaml file
+        nos = "artifact"
+        newpath, tmp_yaml = generate_oci_download_yml(file, tag="1.0", name_on_server=nos, username="wolfv")
+        # Download using this YML file
+        download_oci_file(powerloader_binary, tmp_yaml, file)
+
+        # raise Exception("Stop here!")
```
Comment (wolfv) [inline test/test_OCImain_branch.py:48]: can we remove these commetns?

File: environment.yml (hunk 1 above)
Comment (chGoodchild) [reply environment.yml:23]: not anymore, I will remove it...

File: src/mirrors/oci.cpp (hunk 2 above)
Comment (chGoodchild) [reply src/mirrors/oci.cpp:191]: yep, sorry..

File: src/uploader/oci_upload.cpp (hunk 3 above)
Comment (chGoodchild) [reply src/uploader/oci_upload.cpp:57]: yes, this also...

File: test/remote_mirrors.yml (hunk 4 above)
Comment (chGoodchild) [reply test/remote_mirrors.yml:5]: Ya, it's the same domain... :-/
The intention was redundancy...

File: test/test_OCImain_branch.py (hunk 5 above)
Comment (chGoodchild) [reply test/test_OCImain_branch.py:48]: Sure :)
the exception was just for debugging...

**Pull request merged**